### ISSUE TITLE
Add `result_and_then_ok` lint

### ISSUE THREAD

**Issue by flip1995** [rust-lang/rust-clippy#4412]
In #4386 the `option_and_then_some` lint was added. This can also be implemented for `result_and_then_ok`:

```rust
fn foo() -> Result<(), ()> {
    let x = Ok(10);
    x.and_then(Ok); // This is a noop -> `x`
    x.and_then(|x| Some(..)); // This should be a `map` -> `x.map(|x| ..)`
}
```

_Originally posted by @flip1995 in https://github.com/rust-lang/rust-clippy/pull/4386_

**Comment by printfn** [comment]
I'd like to try and implement this.

**Comment by printfn** [comment]
I've created a pull request #4415.

**Comment by alex-700** [comment]
fixed in #5529
